shared/presentation/src/commonMain/kotlin/network/bisq/mobile/client/ClientMainPresenter.kt

```diff
@@ -1,24 +1,42 @@
 package network.bisq.mobile.client
 
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import network.bisq.mobile.client.websocket.WebSocketClient
+import network.bisq.mobile.domain.data.BackgroundDispatcher
 import network.bisq.mobile.domain.data.repository.main.bootstrap.ApplicationBootstrapFacade
 import network.bisq.mobile.domain.service.market_price.MarketPriceServiceFacade
 import network.bisq.mobile.domain.service.offerbook.OfferbookServiceFacade
 import network.bisq.mobile.presentation.MainPresenter
 
 class ClientMainPresenter(
     private val applicationBootstrapFacade: ApplicationBootstrapFacade,
+    private val webSocketClient: WebSocketClient,
     private val offerbookServiceFacade: OfferbookServiceFacade,
-    private val  marketPriceServiceFacade: MarketPriceServiceFacade
+    private val marketPriceServiceFacade: MarketPriceServiceFacade
 ) : MainPresenter() {
+    private val coroutineScope = CoroutineScope(BackgroundDispatcher)
 
     override fun onViewAttached() {
         super.onViewAttached()
-        applicationBootstrapFacade.activate()
-        offerbookServiceFacade.activate()
-        marketPriceServiceFacade.activate()
+        runCatching {
+            coroutineScope.launch { webSocketClient.connect() }
+
+            applicationBootstrapFacade.activate()
+            offerbookServiceFacade.activate()
+            marketPriceServiceFacade.activate()
+        }.onFailure { e ->
+            // TODO give user feedback (we could have a general error screen covering usual
+            //  issues like connection issues and potential solutions)
+            log.e("Error at onViewAttached", e)
```

Comment:
great comment here, best UX is to inform the user connectivity got lost and automatically try to recover it and announce when we got it back (generally done with a invisible/visble banner under top bar). We'll create an issue to implement this soon.

It would be great for this purpose to have a ping/pong implemented between client and server, so we can calculate latency, inform if there is no response from the server, and things like that.

Reply:
Yes, the network connectivity needs more work anyway and we need to test all the cases when app gets in background, lose internet connection, gets killed by OS,...
I guess that will be quite some effort and would prefer to delegate it for a later stage. And would like to have Tor support first, as that will be the more problematic mode. With clearnet the connection is very fast established but with Tor it will take a few seconds, so the UX impact need to be handled more specific.

I was reading up about Kotlins exception handling strategy, which differs from Java. From my understanding its best to catch exceptions at the layer where we can do something, which is mostly the user layer. Sometimes we might to be able to try to recover from it by e.g. doing automatic reconnects, etc.... And sometimes its useful for debugging and logging to catch and re-throw/repackage an exception. But I guess its better to keep the code relatively clean from exception handling. In Java the checked exceptions can be quite a pain.... I like Kotlins approach better here.

Reply:
that's correct. I think that core components need to "handle their own concerns" if that makes sense. Obviously if it makes sense to do so (to inform client code) you can rethrow / package and rethrow and exception.
Sounds good Henrik!! For sure we'll test the hell out of it in all apps in all the situations we can think of / recreate.